### PR TITLE
provider/aws: Add ability to do cross-region encrypted RDS read replica

### PR DESCRIPTION
**Pull request by JoshiiSinfield**
This should address PR #11784 & #11797

### CODE REVIEW

**Comment by stack72** [comment]
Hi @JoshiiSinfield 

Thanks for the work here - there are a few issues. Firstly, the tests (in the current state) look as follows:

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSDBInstanceReplicaCrossRegion'               1 ↵ ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/02/09 12:39:27 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSDBInstanceReplicaCrossRegion -timeout 120m
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionEncrypted (0.01s)
	testing.go:265: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"aws_db_instance.replica: : invalid or unknown key: destination_region"}
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted (0.01s)
	testing.go:265: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"aws_db_instance.replica: : invalid or unknown key: destination_region"}
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	0.036s
make: *** [testacc] Error 1
```

When i added the parameter to the schema, I then got the second set of errors:

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSDBInstanceReplicaCrossRegion'                 ✹ ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/02/09 12:41:59 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSDBInstanceReplicaCrossRegion -timeout 120m
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionEncrypted (28.28s)
	testing.go:265: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_db_instance.bar: 1 error(s) occurred:

		* aws_db_instance.bar: Error creating DB Instance: InvalidParameterCombination: DB Instance class db.t1.micro does not support encryption at rest
			status code: 400, request id: 43e15e97-eec5-11e6-8165-37af717ea49e
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted (909.19s)
	testing.go:265: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_db_instance.replica: 1 error(s) occurred:

		* aws_db_instance.replica: Error creating DB Instance: MissingRegion: could not find region configuration
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	937.495s
make: *** [testacc] Error 1
```

Were you able to successfully run the tests?

Thanks

Paul

**Comment by JoshiiSinfield** [comment]
HI @stack72 

Thanks for checking the code. Apologies - my first time with Go, so got confused. Although I thought I had I obviously didn't run the tests properly... I've made a couple of amendments and am running them through now.

Cheers,
Josh

**Comment by stack72** [comment]
<3 for the work @JoshiiSinfield :) I know what these tests are like! Please do let me know when you need another review

**Comment by JoshiiSinfield** [comment]
Morning @stack72 

Further reading of the SDK shows I need to send a PreSignedURL - I think that's the last thing I'm missing now. I don't suppose you know of that being done already anywhere in the code do you? 😄

File: builtin/providers/aws/resource_aws_db_instance_test.go [hunk 1]
```diff
@@ -711,6 +753,118 @@ func testAccReplicaInstanceConfig(val int) string {
...
+
+		parameter_group_name = "default.mysql5.6"
+
+		storage_encrypted = "true"
+	}
+
+	resource "aws_db_instance" "replica" {
+		identifier = "tf-replica-db-%d"
+		backup_retention_period = 0
+		replicate_source_db = "${aws_db_instance.bar.identifier}"
```
Comment (bje) [inline builtin/providers/aws/resource_aws_db_instance_test.go]: Should this not be "${aws_db_instance.bar.arn}" ?

**Comment by JoshiiSinfield** [comment]
Hi @stack72 

I've pushed lots more changes up... Could you provide any assistance with like 391? ... I'm struggling to find a way to get access to a Config resource already constructed in order to pass to the aws v4 signer... Any ideas?

Cheers,
Josh

Here's the error below: 

```
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionEncrypted
panic: interface conversion: interface {} is *aws.AWSClient, not *aws.Config

goroutine 673 [running]:
panic(0x18ce1a0, 0xc42086e980)
	/usr/local/Cellar/go/1.7.5/libexec/src/runtime/panic.go:500 +0x1a1
github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsDbInstanceCreate(0xc42089c360, 0x1677dc0, 0xc420170000, 0x3700, 0xdb71)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_db_instance.go:391 +0x69e1
github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc420280d20, 0xc420613d60, 0xc42049ca20, 0x1677dc0, 0xc420170000, 0xc42086e901, 0x37, 0x0)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/helper/schema/resource.go:162 +0x30e
github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc42033e2a0, 0xc4203b57c0, 0xc420613d60, 0xc42049ca20, 0x1, 0x74e602, 0x3700)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/helper/schema/provider.go:212 +0x9b
github.com/hashicorp/terraform/terraform.(*EvalApply).Eval(0xc4208722c0, 0x2d42fe0, 0xc42037c540, 0x2, 0x2, 0x1baac9d, 0x4)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/eval_apply.go:57 +0x24e
github.com/hashicorp/terraform/terraform.EvalRaw(0x2d35f80, 0xc4208722c0, 0x2d42fe0, 0xc42037c540, 0x0, 0x0, 0x0, 0x0)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x17f
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc4204663e0, 0x2d42fe0, 0xc42037c540, 0x2, 0x2, 0x1baac9d, 0x4)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:14 +0xc1
github.com/hashicorp/terraform/terraform.EvalRaw(0x2d368c0, 0xc4204663e0, 0x2d42fe0, 0xc42037c540, 0x2d368c0, 0xc4204663e0, 0xc420466580, 0x17)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x17f
github.com/hashicorp/terraform/terraform.Eval(0x2d368c0, 0xc4204663e0, 0x2d42fe0, 0xc42037c540, 0xc4204663e0, 0x2d368c0, 0xc4204663e0, 0x10)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/eval.go:34 +0x53
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x1b3e000, 0xc42053b578, 0x0, 0x0)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/terraform/graph.go:126 +0xe7c
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc4207338f0, 0x1b3e000, 0xc42053b578, 0xc4206da5c0)
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/dag/walk.go:387 +0x388
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/Users/joshua.sinfield/git/go-work/src/github.com/hashicorp/terraform/dag/walk.go:310 +0xba7
exit status 2
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	545.022s
make: *** [testacc] Error 1

```

File: builtin/providers/aws/resource_aws_db_instance_test.go (hunk 1 above)
Comment (JoshiiSinfield) [inline builtin/providers/aws/resource_aws_db_instance_test.go]: Hi @bje .Thanks! yes it should for a cross-region! I believe it works as identifier for the same region though.

**Comment by radeksimko** [comment]
> Further reading of the SDK shows I need to send a PreSignedURL - I think that's the last thing I'm missing now. I don't suppose you know of that being done already anywhere in the code do you?

I believe you should be able to get it roughly like this:

```go
        req, out := conn.CreateDBInstanceReadReplicaRequest(opts)
        preSignedUrl, err := req.Presign(expireTime)
        if err != nil {
...
```

it shouldn't require re-authentication and duplication of the auth logic inside the resource at all (looking at https://github.com/hashicorp/terraform/pull/11803/files#diff-0b21e21c91dc560c837c95806e2ba913R391).

**Comment by JoshiiSinfield** [comment]
Closer but still not there. I feel my lack of knowledge & experience with Go & the SDK are getting in the way now!

```
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSDBInstanceReplicaCrossRegion -timeout 120m
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionEncrypted (801.30s)
	testing.go:265: Step 0 error: Error applying: 1 error(s) occurred:
		
		* aws_db_instance.replica: 1 error(s) occurred:
		
		* aws_db_instance.replica: Error creating DB Instance: InvalidParameterValue: PreSignedUrl could not be authenticated.
			status code: 400, request id: 20bde221-f35a-11e6-a2cc-734feb2b6c9a
	testing.go:329: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.
		
		Error: Error applying: 1 error(s) occurred:
		
		* aws_vpc.vpc (destroy): 1 error(s) occurred:
		
		* aws_vpc.vpc: DependencyViolation: The vpc 'vpc-1c4de074' has dependencies and cannot be deleted.
			status code: 400, request id: 2d40a944-9b54-4359-a269-e8c08ecda89b
		
		State: aws_vpc.vpc:
		  ID = vpc-1c4de074
		  provider = aws.eu
		  cidr_block = 172.18.0.0/22
		  default_network_acl_id = acl-1343ca7b
		  default_route_table_id = rtb-4f801127
		  default_security_group_id = sg-64c42c0f
		  dhcp_options_id = dopt-83d531eb
		  enable_dns_hostnames = true
		  enable_dns_support = true
		  instance_tenancy = default
		  main_route_table_id = rtb-4f801127
		  tags.% = 1
		  tags.Name = bar-test-vpc-7581855244856810280
=== RUN   TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionUnEncrypted (2001.23s)
	testing.go:265: Step 0 error: Check failed: Check 1/3 error: DBInstanceNotFound: DBInstance foobaz-test-terraform-7544857979864918209 not found.
			status code: 404, request id: 651466d3-f35e-11e6-bc47-f9bbf08af73a
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	2802.573s
make: *** [testacc] Error 1

```

**Comment by JoshiiSinfield** [comment]
little closer still (sorry for frequent small updates...struggling - should have left it to adults!)

```
--- FAIL: TestAccAWSDBInstanceReplicaCrossRegionEncrypted (761.44s)
    testing.go:265: Step 0 error: Error applying: 1 error(s) occurred:
        
        * aws_db_instance.replica: 1 error(s) occurred:
        
        * aws_db_instance.replica: Error creating DB Instance: InvalidParameterValue: PreSignedUrl could not be authenticated.
            status code: 400, request id: 307066d9-f373-11e6-bd82-61b99b03590f
FAIL
exit status 1
FAIL    github.com/hashicorp/terraform/builtin/providers/aws    761.471s
make: *** [testacc] Error 1
```

This is what CloudTrail shows from the API call:

```
{
    "eventVersion": "1.04",
    "userIdentity": {
        "type": "IAMUser",
        "principalId": "****",
        "arn": "arn:aws:iam::****:user/****",
        "accountId": "****",
        "accessKeyId": "****",
        "userName": "****"
    },
    "eventTime": "2017-02-15T11:37:57Z",
    "eventSource": "rds.amazonaws.com",
    "eventName": "CreateDBInstanceReadReplica",
    "awsRegion": "eu-central-1",
    "sourceIPAddress": "****",
    "userAgent": "APN/1.0 HashiCorp/1.0 Terraform/0.9.0-dev",
    "errorCode": "InvalidParameterValueException",
    "errorMessage": "PreSignedUrl could not be authenticated.",
    "requestParameters": {
        "dBInstanceClass": "db.t2.large",
        "publiclyAccessible": false,
        "dBSubnetGroupName": "foobarbaz-test-161470128570676312",
        "copyTagsToSnapshot": false,
        "dBInstanceIdentifier": "tf-replica-db-161470128570676312",
        "tags": [
            {
                "value": "tf-replica-db-161470128570676312",
                "key": "Name"
            }
        ],
        "sourceDBInstanceIdentifier": "arn:aws:rds:us-west-2:******:db:foobarbaz-test-terraform-161470128570676312",
        "kmsKeyId": "arn:aws:kms:eu-central-1:*****:key/********-****-****-****-efa9040db46d"
    },
    "responseElements": null,
    "requestID": "307066d9-f373-11e6-bd82-61b99b03590f",
    "eventID": "612d6e65-db21-4024-b8df-a7e1bfe239fb",
    "eventType": "AwsApiCall",
    "recipientAccountId": "******"
}
```

```
2017/02/15 12:32:32 [DEBUG] DB Instance Replica create configuration: {
  CopyTagsToSnapshot: false,
  DBInstanceClass: "db.t2.large",
  DBInstanceIdentifier: "tf-replica-db-4642037333531415180",
  DBSubnetGroupName: "foobarbaz-test-4642037333531415180",
  DestinationRegion: "eu-central-1",
  KmsKeyId: "arn:aws:kms:eu-central-1:*********:key/******-****-****-****",
  PreSignedUrl: "https://rds.eu-central-1.amazonaws.com/?Action=CreateDBInstanceReadReplica&CopyTagsToSnapshot=false&DBInstanceClass=db.t2.large&DBInstanceIdentifier=tf-replica-db-4642037333531415180&DBSubnetGroupName=foobarbaz-test-4642037333531415180&DestinationRegion=eu-central-1&KmsKeyId=arn%3Aaws%3Akms%3Aeu-central-1%3A*********%3Akey%2F******-****-****-****&PubliclyAccessible=false&SourceDBInstanceIdentifier=arn%3Aaws%3Ards%3Aus-west-2%3A*********%3Adb%3Afoobarbaz-test-terraform-4642037333531415180&Tags.member.1.Key=Name&Tags.member.1.Value=tf-replica-db-4642037333531415180&Version=2014-10-31&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=*********%2F20170215%2Feu-central-1%2Frds%2Faws4_request&X-Amz-Date=20170215T123232Z&X-Amz-Expires=300&X-Amz-SignedHeaders=host&X-Amz-Signature=08a6d57537b8c0cade719cea06f9c36853c45db7876aef90bac8329feb11ba12",
  PubliclyAccessible: false,
  SourceDBInstanceIdentifier: "arn:aws:rds:us-west-2:*********:db:foobarbaz-test-terraform-4642037333531415180",
  SourceRegion: "us-west-2",
  Tags: [{
      Key: "Name",
      Value: "tf-replica-db-4642037333531415180"
    }]
}
2017/02/15 12:32:32 [DEBUG] [aws-sdk-go] DEBUG: Request rds/CreateDBInstanceReadReplica Details:
---[ REQUEST POST-SIGN ]-----------------------------
GET /?Action=CreateDBInstanceReadReplica&CopyTagsToSnapshot=false&DBInstanceClass=db.t2.large&DBInstanceIdentifier=tf-replica-db-4642037333531415180&DBSubnetGroupName=foobarbaz-test-4642037333531415180&DestinationRegion=eu-central-1&KmsKeyId=arn%3Aaws%3Akms%3Aeu-central-1%3A*********%3Akey%2F******-****-****-****&PubliclyAccessible=false&SourceDBInstanceIdentifier=arn%3Aaws%3Ards%3Aus-west-2%3A*********%3Adb%3Afoobarbaz-test-terraform-4642037333531415180&Tags.member.1.Key=Name&Tags.member.1.Value=tf-replica-db-4642037333531415180&Version=2014-10-31&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=*********%2F20170215%2Feu-central-1%2Frds%2Faws4_request&X-Amz-Date=20170215T123232Z&X-Amz-Expires=300&X-Amz-SignedHeaders=host&X-Amz-Signature=08a6d57537b8c0cade719cea06f9c36853c45db7876aef90bac8329feb11ba12 HTTP/1.1
Host: rds.eu-central-1.amazonaws.com
User-Agent: APN/1.0 HashiCorp/1.0 Terraform/0.9.0-dev
Accept-Encoding: gzip


-----------------------------------------------------
2017/02/15 12:32:36 [DEBUG] [aws-sdk-go] DEBUG: Response rds/CreateDBInstanceReadReplica Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 400 Bad Request
Connection: close
Content-Length: 292
Content-Type: text/xml
Date: Wed, 15 Feb 2017 12:32:33 GMT
X-Amzn-Requestid: d1a26870-f37a-11e6-85a8-55eb6eeaef7b

<ErrorResponse xmlns="http://rds.amazonaws.com/doc/2014-10-31/">
  <Error>
    <Type>Sender</Type>
    <Code>InvalidParameterValue</Code>
    <Message>PreSignedUrl could not be authenticated.</Message>
  </Error>
  <RequestId>d1a26870-f37a-11e6-85a8-55eb6eeaef7b</RequestId>
</ErrorResponse>

-----------------------------------------------------
2017/02/15 12:32:36 [DEBUG] DB Instance Replica response output: {

}
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalWriteState
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalIf
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalWriteState
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalWriteDiff
2017/02/15 12:32:36 [DEBUG] root: eval: *terraform.EvalApplyPost
2017/02/15 12:32:36 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_db_instance.replica: Error creating DB Instance: InvalidParameterValue: PreSignedUrl could not be authenticated.
    status code: 400, request id: d1a26870-f37a-11e6-85a8-55eb6eeaef7b
2017/02/15 12:32:36 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_db_instance.replica: Error creating DB Instance: InvalidParameterValue: PreSignedUrl could not be authenticated.
    status code: 400, request id: d1a26870-f37a-11e6-85a8-55eb6eeaef7b
```

So... From what I can see the PreSignedUrl is being signed in the destination region, not the source region. If i read the SDK docs correctly it should be like https://rds.us-west-2.amazonaws.com/?... ... not eu-central-1.

**Comment by stack72** [comment]
FYI, opened an issue with the SDK to try and work out what is going wrong

I have been able to replicate this perfectly 

https://github.com/aws/aws-sdk-go/issues/1098

**Comment by drubin** [comment]
Thanks for doing work on this. I assume we will be able to port this into a `0.8.8` release?  as currently after upgrading to 0.8.7 we are unable to create read replica's any more and would ideally like to stick to the 0.8 until 0.9 is fully released?

**Comment by JoshiiSinfield** [comment]
Hi @stack72 ,

Thanks for raising an issue with the SDK. I've just been reading it - it seems that too many options are being set? 

did you get the replica created in eu-central-1, encrypted? Sounds good if so 😄

**Comment by stack72** [comment]
Hi @JoshiiSinfield 

So i am still working on this - it's pretty nasty code right now - I promise i will shout you when it's ready to rock - should be another day or 2 max

P.

**Comment by jessecollier** [comment]
patiently waiting for this to merge, impatiently ;)

This is currently blocking us from upgrading past `0.8.5` so deff a 👍  on this.

**Comment by jdn-za** [comment]
@jessecollier in the same boat, had to downgrade all our envs to 0.8.5

**Comment by drubin** [comment]
@stack72 @JoshiiSinfield Any progress on this one? I saw 0.9 was released and we would LOVE to upgrade and test it out but this is currently a major blocker.

**Comment by drubin** [comment]
Does this fix this issue as well https://github.com/hashicorp/terraform/pull/12680

**Comment by JoshiiSinfield** [comment]
@drubin  No progress from myself... @stack72 seemed to be making headway with it. If Paul would like I'd be happy to take another look now we have further information. He did however start #12241 following this PR, so maybe he wants to take it in a different direction?

**Comment by stack72** [comment]
Hi Friends

this was stalled due to 0.9 release - i will be going back to this work very shortly. We are going to take a new direction here

We are going to split the replica work to a new resource

aws_db_instance
aws_db_instance_replica

This will mean that we can start to share the schema and be able to have it much more thoroughly tested

Paul

**Comment by JoshiiSinfield** [comment]
@stack72 Thanks for the update.

Sounds a great plan. Let me know if there's anything I can help with.

Cheers,
Josh

**Comment by drubin** [comment]
@stack72 Thanks for the update if you would like us to test any of the changes I am happy to give it a go and give feedback. Currently because of this issue we are unable to upgrade past 0.8.5 so anything that makes it easier to upgrade would be great.

**Comment by bclodius** [comment]
Looking forward to this one!

**Comment by zerolaser** [comment]
@stack72  when will this change be added to terraform ?

**Comment by kuldeepmarganache** [comment]
Hello @stack72, thank you for the work. When will this fix be included in terraform?

**Comment by chnaveens** [comment]
@stack72 May i know when are we going to get aws_db_instance_replica resource. We badly need this ASAP...!

**Comment by bclodius** [comment]
@kuldeepmarganache @chnaveens I've looked at the cross-region API's and I think the reason this PR is stalled is because the creation API's don't allow the user to specify many of the required parameters like SecurityGroups, ParameterGroup, etc.  I believe the cross region encrypted read replica was probably rushed by AWS and currently the flow is only friendly via their console.

**Comment by apparentlymart** [comment]
Hello @JoshiiSinfield, and thanks for working on this!

As part of the the Terraform 0.10 release earlier this year, all of the Terraform providers were moved to their own repositories in [the terraform-providers GitHub organization](https://github.com/terraform-providers), and removed from the Terraform Core repository.

Unfortunately due to the fact that new issues and pull requests are being opened constantly, it was not possible for the various provider maintainers to merge all outstanding pull requests before this split, and there is no automatic way to migrate a pull request to a new repository.

As a result, this pull request can sadly no longer be applied as-is, and so I'm going to close it.

If you or someone else has the time and motivation to apply same changes to [the `aws` provider repository](https://github.com/terraform-providers/terraform-provider-aws) and open a new PR there -- assuming that the blocking issues discussed above were resolved in the mean time -- the maintainers of that provider should be able to review and merge it.

Thanks again for working on this, and sorry it was not able to be merged before the provider repository changes.

**Pull request closed without merging**